databricks/koalas/groupby.py

```diff
@@ -418,6 +419,81 @@ def size(self):
                                              for i, s in enumerate(groupkeys)])
         return _col(DataFrame(internal))
 
+    def diff(self, periods=1):
+        """
+        First discrete difference of element.
+
+        Calculates the difference of a DataFrame element compared with another element in the
+        DataFrame (default is the element in the same column of the previous row).
+
+        .. note:: the current implementation of diff uses Spark's Window without
+            specifying partition specification. This leads to move all data into
+            single partition in single machine and could cause serious
+            performance degradation. Avoid this method against very large dataset.
+
+        Parameters
+        ----------
+        periods : int, default 1
+            Periods to shift for calculating difference, accepts negative values.
+
+        Returns
+        -------
+        diffed : DataFrame or Series
+
+        See Also
+        --------
+        databricks.koalas.Series.groupby
+        databricks.koalas.DataFrame.groupby
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'a': [1, 2, 3, 4, 5, 6],
+        ...                    'b': [1, 1, 2, 3, 5, 8],
+        ...                    'c': [1, 4, 9, 16, 25, 36]}, columns=['a', 'b', 'c'])
+        >>> df
+           a  b   c
+        0  1  1   1
+        1  2  1   4
+        2  3  2   9
+        3  4  3  16
+        4  5  5  25
+        5  6  8  36
+
+        >>> df.groupby(['b']).diff()
+             a    c
+        0  NaN  NaN
+        1  1.0  3.0
+        2  NaN  NaN
+        3  NaN  NaN
+        4  NaN  NaN
+        5  NaN  NaN
+
+        Difference with previous column
+
+        >>> df.groupby(['b'])['a'].diff()
+        0    NaN
+        1    1.0
+        2    NaN
+        3    NaN
+        4    NaN
+        5    NaN
+        Name: a, dtype: float64
+        """
+        sdf = self._kdf._sdf
+        index = self._kdf._internal.index_columns[0]
+        for i, s in enumerate(self._agg_columns):
+            window = Window.partitionBy([s._scol for s in self._groupkeys])\
```

Comment:
This one too. Can we deduplicate with `Series.diff`?

Reply:
I think there is no need to deduplicate with `Series.diff`.Unlike `Series.diff`, there is `partitionBy ` in `groupby.diff`.

Reply:
you can do it like https://github.com/databricks/koalas/commit/429e26be5bbbf90f7887cfc5531ec1210a7f5f11 or https://github.com/databricks/koalas/commit/510cad590461e5ea42fb2e2a4846f62ae55a210f